### DISCUSSION TITLE
Endpoint access: url params format

### DISCUSSION THREAD

**Discussion by IamSAB** [directus/directus#3943] [Archive]
I want to convert the query below to url params. I am using fetch and not axios (the sdk js), so I have to generate the url myself.

I got an events collection where I want to retrieve running and upcoming events (means events in a range "from now" to "in two months").
```javascript
filter: {
  _or: [{
    _and: [{
      from: {
        _lte: rangeStart
      }
    },
    {
      to: {
        _gte: rangeEnd
      }
    }]
  },
  {
    from: {
      _between: dateRange,
    },
  },
  {
    to: {
      _between: dateRange,
    },
  }
}],
```
Flattening the object to this params will throw an error:
```
filter[_or][0][_and][0][from][_lte]=rangeStart
filter[_or][0][_and][1][to][_gte]=rangeEnd
filter[_or][1][from][_between]=dateRange
filter[_or][2][to][_between]=dateRange
```
I also tried to remove the indexes, also will not work:
```
filter[_or][][_and][][from][_lte]=rangeStart
filter[_or][][_and][][to][_gte]=rangeEnd
filter[_or][][from][_between]=dateRange
filter[_or][][to][_between]=dateRange
```

Here the error as reference:
```json
"select \"events\".\"title\", \"events\".\"id\" from \"events\" where (((\"\" = $1 and \"\" = $2) or \"events\".\"from\" between $3 and $4 or \"events\".\"to\" between $5 and $6)) limit $7 - Bezeichner in Anführungszeichen hat Länge null bei »\"\"«"
```

What I know is, that the second part, gives correct results (independent of using indexes or not):
```
filter[_or][][from][_between]=dateRange
filter[_or][][to][_between]=dateRange
```
So I concluded, there is either a problem with my syntax (or better my conversion) or with the resolving of nested AND or OR's by directus. Can somebody help me out?

*And yes, the query works: just serializing the object and pass the string as filter.  Then why I want to do the above conversion? I'd like to use the same format for every query parameter and do not want to check what I should json serialize and what not.*

**Comment by benhaynes** [comment]
I _think_ @rijkvanzanten is writing updated docs now for exactly this... :)

**Reply by rijkvanzanten** [reply]
@IamSAB The query param also accepts JSON, so using

```js
filter: JSON.stringify({
  _or: [{
    _and: [{
      from: {
        _lte: rangeStart
      }
    },
    {
      to: {
        _gte: rangeEnd
      }
    }]
  },
  {
    from: {
      _between: dateRange,
    },
  },
  {
    to: {
      _between: dateRange,
    },
  }
}]);
```

should work!
